app/work_template_executor.go

```diff
@@ -0,0 +1,277 @@
+	cleanChannelName := cleanChannelName(channelDisplayName)
+	channelName := cleanChannelName
+	if len(channelName) > model.ChannelNameMaxLength {
+		channelName = channelName[:model.ChannelNameMaxLength]
+	}
+
+	// Mostly because of the "quick use" feature, we might try to create channel that have the exact same "Name"
+	// This loop ensures that if the original name is taken, we try again by adding a suffix to the Name
+	for channelCreationAppErr != nil {
+		// create channel
+		newChan, channelCreationAppErr := e.app.CreateChannelWithUser(c, &model.Channel{
+			TeamId:      wtcr.TeamID,
+			Name:        channelName,
+			DisplayName: channelDisplayName,
+			Type:        model.ChannelTypeOpen,
+			Purpose:     cChannel.Purpose,
+			Props: map[string]any{
+				model.WorkTemplateIDChannelProp: wtcr.WorkTemplate.ID,
+			},
+		}, c.Session().UserId)
+		if channelCreationAppErr != nil {
+			if channelCreationAppErr.Id == store.ChannelExistsError {
+				// compute a new unique name
+				suffix := fmt.Sprintf("-%s", model.NewId()[0:4])
+				channelName = cleanChannelName
+				if len(cleanChannelName)+len(suffix) > model.ChannelNameMaxLength {
+					channelName = cleanChannelName[:model.ChannelNameMaxLength-len(suffix)]
+				}
+				channelName = channelName + suffix
+				continue
+			}
+
+			return "", fmt.Errorf("error while creating channel: %w", channelCreationAppErr)
+		}
+		channelID = newChan.Id
+	}
+
+	return channelID, nil
+}
+
+func (e *appWorkTemplateExecutor) CreateBoard(
+	c *request.Context,
+	wtcr *worktemplates.ExecutionRequest,
+	cBoard *model.WorkTemplateBoard,
+	linkToChannelID string,
+) (string, error) {
+	boardService := e.app.Srv().services[product.BoardsKey].(product.BoardsService)
+	templates, err := boardService.GetTemplates("0", c.Session().UserId)
+	if err != nil {
+		return "", fmt.Errorf("error while getting templates: %w", err)
+	}
+
+	var template *fb_model.Board = nil
+	for _, t := range templates {
+		v, ok := t.Properties["trackingTemplateId"]
+		if ok && v == cBoard.Template {
+			template = t
+			break
+		}
+	}
+	if template == nil {
+		return "", fmt.Errorf("template not found")
+	}
+
+	title := cBoard.Name
+	if wtcr.Name != "" {
+		title = fmt.Sprintf("%s: %s", wtcr.Name, cBoard.Name)
+	}
+
+	// Duplicate board From template
+	boardsAndBlocks, _, err := boardService.DuplicateBoard(template.ID, c.Session().UserId, wtcr.TeamID, false)
+	if err != nil {
+		return "", fmt.Errorf("failed to create new board from template: %w", err)
+	}
+	if len(boardsAndBlocks.Boards) != 1 {
+		return "", fmt.Errorf("only one board was expected, found %d", len(boardsAndBlocks.Boards))
+	}
+
+	// Apply patch for the title and linked channel
+	_, err = boardService.PatchBoard(&fb_model.BoardPatch{
+		Title:     &title,
+		ChannelID: &linkToChannelID,
+	}, boardsAndBlocks.Boards[0].ID, c.Session().UserId)
+	if err != nil {
+		return "", fmt.Errorf("failed to patch board: %w", err)
+	}
+
+	return boardsAndBlocks.Boards[0].ID, nil
```

Comment:
Is it ok that you're returning the unpatched board here?

Reply:
We just want to return the board ID, so unless the patch leads to a new ID we should be good.

Reply:
I guess for the sake of being safe and future proof I'll change to code to use the patched board it.